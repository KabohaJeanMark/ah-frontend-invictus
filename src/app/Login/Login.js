import React, { Component } from 'react';

class Login extends Component {
  render () {
    return (
      <div>
        <h5>Authors Haven Login</h5>
      </div>
    );
  }
}

export default Login;
