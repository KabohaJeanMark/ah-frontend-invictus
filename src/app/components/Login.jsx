import React from 'react';
import PropTypes from 'prop-types';
import '../../assets/styles/login.scss';

const Login = (
  {
    handleSubmit,
    onChange,
    email,
    password,
    errors
  }
) => {
  return (
    <div className="row">
      <div className="col-sm-4" />
      <div className="col-sm-4">
        <div className="banner">
          <img className="banner-img"
            src={require("../../assets/images/ah_banner.png")}
            alt={"Banner"} />
        </div>
        <div className="card">
          <form onSubmit={handleSubmit}>
            <div className="form-group">
              <label>Login Form</label>
              <input type="text" id="UserEmail" onChange={onChange} value={email || ''} name="email"
                className="form-control input-field" placeholder="Email" required />
              {
                errors.userEmail && (<div style={{ color: 'red' }}>{errors.userEmail}</div>)
              }
              <input type="password" onChange={onChange} value={password} name="password"
                className="form-control input-field" placeholder="Password" required />
              {
                errors.userPassword && (<div style={{ color: 'red' }}>{errors.userPassword}</div>)
              }
              <input type="submit" className="btn btn-primary login-btn" />
              <p className="have-account">Forgot password? <a href="/password/reset">Reset</a></p>
            </div>
            <div className="loginwith">
              <img className="loginwith-img" alt="login-with"
                src={require("../../assets/images/loginwith.png")} />
              <div className="row social-media-row">
                <div className="col-md-2" />
                <div className="col-md-8">
                  <img className="socialmedia-img facebook-img" alt="facebook-logo"
                    src={require("../../assets/images/facebook.png")} />
                  <img className="socialmedia-img google-img" alt="google-logo"
                    src={require("../../assets/images/google.png")} />
                  <img className="socialmedia-img twitter-img" alt="twitter-logo"
                    src={require("../../assets/images/twitter.png")} />
                </div>
                <div className="col-md-2" />
              </div>
            </div>
            <p className="have-account">Not yet a member? <a href="/signup">Sign up</a></p>
          </form>
        </div>
      </div>
      <div className="col-sm-3" />
    </div>
  );
};
Login.propTypes = {
  email: PropTypes.string,
  password: PropTypes.string,
  errors: PropTypes.object,
  handleSubmit: PropTypes.func,
  onChange: PropTypes.func
};

export default Login;
