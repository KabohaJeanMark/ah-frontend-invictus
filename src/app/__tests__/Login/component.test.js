import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import Login from '../../components/Login';

describe('<Login />', () => {
  let wrapper;
  const props = {
    handleSubmit: jest.fn(),
    onChange: jest.fn(),
    email: '',
    password: '',
    errors: {}
  };
  it('renders the login page', () => {
    wrapper = shallow(<Login {...props} />);
  });

  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
