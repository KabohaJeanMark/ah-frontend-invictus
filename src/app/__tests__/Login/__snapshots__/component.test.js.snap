// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> matches snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col-sm-4"
  />
  <div
    className="col-sm-4"
  >
    <div
      className="banner"
    >
      <img
        alt="Banner"
        className="banner-img"
        src="ah_banner.png"
      />
    </div>
    <div
      className="card"
    >
      <form
        onSubmit={[MockFunction]}
      >
        <div
          className="form-group"
        >
          <label>
            Login Form
          </label>
          <input
            className="form-control input-field"
            id="UserEmail"
            name="email"
            onChange={[MockFunction]}
            placeholder="Email"
            required={true}
            type="text"
            value=""
          />
          <input
            className="form-control input-field"
            name="password"
            onChange={[MockFunction]}
            placeholder="Password"
            required={true}
            type="password"
            value=""
          />
          <input
            className="btn btn-primary login-btn"
            type="submit"
          />
          <p
            className="have-account"
          >
            Forgot password? 
            <a
              href="/password/reset"
            >
              Reset
            </a>
          </p>
        </div>
        <div
          className="loginwith"
        >
          <img
            alt="login-with"
            className="loginwith-img"
            src="loginwith.png"
          />
          <div
            className="row social-media-row"
          >
            <div
              className="col-md-2"
            />
            <div
              className="col-md-8"
            >
              <img
                alt="facebook-logo"
                className="socialmedia-img facebook-img"
                src="facebook.png"
              />
              <img
                alt="google-logo"
                className="socialmedia-img google-img"
                src="google.png"
              />
              <img
                alt="twitter-logo"
                className="socialmedia-img twitter-img"
                src="twitter.png"
              />
            </div>
            <div
              className="col-md-2"
            />
          </div>
        </div>
        <p
          className="have-account"
        >
          Not yet a member? 
          <a
            href="/signup"
          >
            Sign up
          </a>
        </p>
      </form>
    </div>
  </div>
  <div
    className="col-sm-3"
  />
</div>
`;
